tests/c_api_tests/test_c_api_meta_data.cpp

```diff
@@ -17,24 +17,24 @@ TEST_CASE("C API Meta Data") {
     // get handle
     HandlePtr const unique_handle{PGM_create_handle()};
     PGM_Handle* hl = unique_handle.get();
-    auto const& meta = meta_data();
+    auto const& meta = meta_data;
 
     SUBCASE("Datasets") {
         // check dataset
         CHECK(PGM_meta_n_datasets(hl) == meta.n_datasets());
         for (Idx idx_dataset = 0; idx_dataset != meta.n_datasets(); ++idx_dataset) {
             PGM_MetaDataset const* const dataset = PGM_meta_get_dataset_by_idx(hl, idx_dataset);
-            char const* const dataset_name = PGM_meta_dataset_name(hl, dataset);
-            CHECK(PGM_meta_get_dataset_by_name(hl, dataset_name) == dataset);
+            std::string const dataset_name = PGM_meta_dataset_name(hl, dataset);
```

Comment:
or just `auto`

```suggestion
            auto const dataset_name = PGM_meta_dataset_name(hl, dataset);
```

(same below)

Reply:
You can't. We have to instantiate a `string`, otherwise it is `char const*`. In the later check, we need to compare the string, not the address of `char const*`.